src/txn/txn_recover.c

```diff
@@ -452,6 +452,69 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the checkpoint snapshot details as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG list;
+    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM k, v;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint8_t counter;
+    char *endptr;
+    char *sys_config;
+
+    sys_config = NULL;
+    session = r->session;
+    conn = S2C(session);
+    counter = 0;
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MIN, &cval));
+        if (cval.len != 0)
+            conn->txn_global.ckpt_snap_min = (uint64_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MAX, &cval));
+        if (cval.len != 0)
+            conn->txn_global.ckpt_snap_max = (uint64_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_COUNT, &cval));
+        if (cval.len != 0)
+            conn->txn_global.ckpt_snapshot_count = (uint32_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT, &cval));
+        if (cval.len != 0) {
+            __wt_config_subinit(session, &list, &cval);
+            WT_ERR(__wt_calloc_def(
+              session, conn->txn_global.ckpt_snapshot_count, &conn->txn_global.ckpt_snapshot));
+            while (__wt_config_next(&list, &k, &v) == 0) {
+                if (k.type == WT_CONFIG_ITEM_NUM)
+                    conn->txn_global.ckpt_snapshot[counter++] =
+                      (uint64_t)strtoll(k.str, &endptr, 10);
+            }
+        }
+        WT_ASSERT(session, conn->txn_global.ckpt_snapshot_count == counter);
```

Comment:
We should do more checking here, if any of the things are present, I think we expect them all to be present? At least MIN/MAX. I suspect it's possible to have a snapshot with 0 entries, but we should be certain and assert as much as we can.

Reply:
I don't think it is necessary to check if MIN/MAX are present because I save all these details into the metadata file [here](https://github.com/wiredtiger/wiredtiger/pull/6029/files#diff-3bf20c92a30d69f9957e610d265f1aa0R982) when the `snapshot_count > 0` and  I expect `snap_min`, `snap_max` and `snapshot` are set appropriately while getting a snapshot.

Reply:
Added assert at a couple of places to make sure all the snapshot details be present in the metadata file.